src/node_platform.cc

```diff
@@ -45,6 +45,19 @@ static void PlatformWorkerThread(void* data) {
   }
 }
 
+static int GetActualThreadPoolSize(int thread_pool_size) {
+  if (thread_pool_size < 1) {
+    uv_cpu_info_t* cpu_info;
+    int count;
+
+    if (uv_cpu_info(&cpu_info, &count) == 0) {
+      uv_free_cpu_info(cpu_info, count);
+      thread_pool_size = count - 1;
+    }
+  }
```

Comment:
I suggest waiting for the upgrade to libuv 1.44 and replacing this with [uv_available_parallelism()](https://docs.libuv.org/en/v1.x/misc.html#c.uv_available_parallelism), it's specifically for use cases such as this one.

`uv_cpu_info()` isn't really appropriate because it doesn't know how many processors are available to the process, only how many are online.

Reply:
`uv_available_parallelism()` looks more suited to this case indeed. I'll wait for the upgrade to land.